locale_linux_test.go

```diff
@@ -4,64 +4,68 @@ import (
 	"errors"
 	"testing"
 
-	"bou.ke/monkey"
 	"github.com/stretchr/testify/assert"
 	"golang.org/x/text/language"
 )
 
+func hookDetects(Env func() string, Locale func() (string, error)) func() {
```

Comment:
Well, I got your idea for now.

But `hookDetects`, `hookLookupEnv` and `hookExecCommand` here look like just a specialization version for [`patchValue`](https://github.com/bouk/monkey/blob/b118a17387657e860bb85a0f3dafd15cad03c266/monkey.go#L67).